oauth2_provider/models.py

```diff
@@ -123,6 +129,21 @@ def clean(self):
     def get_absolute_url(self):
         return reverse('oauth2_provider:detail', args=[str(self.id)])
 
+    def get_allowed_scopes_from_scopes(self, scopes):
+        """
+        Filters the given list of scopes so it only contains the allowed scopes
+        """
+        scopes = scopes.split(' ')
+        allowed_scopes = None
+        if application.allowed_scopes:
```

Comment:
`s/application.allowed_scopes/self.allowed_scopes/`. Excited for this PR to get merged.